spec/api/comment_thread_spec.rb

```diff
@@ -592,7 +592,24 @@ def thread_result(id, params)
         comment.reload
         comment.endorsed.should == false
         comment.endorsement.should == nil
-        check_thread_result_json(nil, changed_thread, parse(last_response.body))
+        json_response = parse(last_response.body)
+        check_thread_result_json(nil, changed_thread, json_response)
+        json_response["read"].should == false
+      end
+      it "update information of comment thread and mark thread as read" do
+        thread = CommentThread.first
+        user = create_test_user(42)
+        put "/api/v1/threads/#{thread.id}", body: "new body", title: "new title", commentable_id: "new_commentable_id", thread_type: "question", read: true, user_id: user.id
+        last_response.should be_ok
+        changed_thread = CommentThread.find(thread.id)
+        changed_thread.body.should == "new body"
+        changed_thread.title.should == "new title"
+        changed_thread.commentable_id.should == "new_commentable_id"
+        changed_thread.thread_type.should == "question"
+        user.mark_as_read(thread)
```

Comment:
Agreed.  If you need line 609 to make this pass it means there is likely a bug, even if it is only in the check_thread_result_json assertion code.